src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -347,42 +322,79 @@ class NotebookEditor extends Component {
       ])
-    }
-
-    const isCreating = clusterStatus === 'Creating'
-    const currentStep = clusterStatus !== 'Running' ? 1 : 2
-
-    const step = (index, text) => div({ style: styles.step.container }, [
-      div({ style: styles.step.col1 }, [
-        index < currentStep && icon('check', { size: 24, style: { color: colors.green[0] } }),
-        index === currentStep && (failed ? icon('times', { size: 24, style: { color: colors.red[0] } }) : spinner())
-      ]),
-      div({ style: styles.step.col2 }, [text])
-    ])
+    } else {
+      const isCreating = clusterStatus === 'Creating'
+      const isRunning = clusterStatus === 'Running'
+      const currentStep = isRunning ? 2 : 1
+
+      const makeStep = (index, shortText, fullText) => {
+        const isCurrent = index === currentStep
+        const isComplete = index < currentStep
+        const isIncomplete = index > currentStep
+        const stepCond = (current, complete, incomplete) => isCurrent ? current : isComplete ? complete : incomplete
+        const baseColor = stepCond(colors.blue, colors.green, colors.gray)
+
+        return div({
+          style: {
+            marginRight: '1rem', padding: '0.5rem 1rem',
+            borderRadius: '1rem', border: `1px solid ${baseColor[0]}`,
+            backgroundColor: stepCond(colors.blue[1], colors.green[1], colors.gray[6]),
+            color: isIncomplete ? undefined : 'white',
+            fontWeight: isCurrent ? 600 : undefined
+          }
+        }, [
+          !isIncomplete && icon(isCurrent ? 'loadingSpinner' : 'success-standard', {
+            className: 'is-solid',
+            style: { marginRight: '0.5rem' }
+          }),
+          isCurrent ? fullText : shortText
+        ])
+      }
 
-    return h(Fragment, [
-      div({ style: styles.pageContainer }, [
-        div({ style: Style.elements.sectionHeader }, ['Terra is preparing your notebook']),
-        step(1,
-          Utils.cond(
-            [clusterError, clusterError],
-            [isCreating, 'Creating notebook runtime'],
-            'Waiting for notebook runtime to be ready'
-          )
-        ),
-        isCreating && div({ style: styles.creatingMessage }, [
-          icon('info', { size: 24, style: { color: colors.orange[0] } }),
-          'This can take several minutes. You may navigate away and come back once the cluster is ready.'
+      return h(Fragment, [
+        (cluster || createRequested) ?
+          div({ style: { padding: '2rem', display: 'flex' } }, [
+            failed ?
+              h(Fragment, [
+                icon('times', { size: 24, style: { color: colors.red[0], marginRight: '1rem' } }),
+                clusterError || 'Error launching notebook.'
+              ]) :
+              h(Fragment, [
+                makeStep(1, 'Runtime started', (isCreating || createRequested) ?
+                  'Creating notebook runtime environment. You can navigate away and return in 5-10 minutes.' :
```

Comment:
```suggestion
                  'Creating notebook runtime environment. You can return in 5-10 minutes.' :
```
Saves button real-estate